### PR TITLE
[PR] - Add JSON dataset support in ADF

### PR DESCRIPTION
**Pull request by shawnxzq**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json [hunk 1]
```diff
@@ -651,6 +651,43 @@
         }
       }
     },
+    "JsonDataset": {
```
Comment (amarzavery) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json]: Please update examples that showcase the new models that have been added.

**Review by amarzavery — COMMENTED** [review]
Adding new child model to an existing polymorphic hierarchy in an existing stable api version will cause breaking changes to the generated SDKs.

**Review by shawnxzq — COMMENTED** [review]
@hvermis I think this is a common practice in data factory. We don't assume customer with lower version of SDK to be able to consume everything generated by a higher version of SDK, right?

**Review by hvermis — APPROVED** [review]

**Comment by hvermis** [comment]
@KrisBash Can you please review this PR? It is again very similar to other ADF PRs that you reviewed in the past - we are adding new Dataset types again.

**Comment by ravbhatnagar** [comment]
Same concern as with the other PR from DataFactory RP. writable properties being added to existing API version is against Azure API guidelines. https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change

**Comment by shawnxzq** [comment]
> Same concern as with the other PR from DataFactory RP. writable properties being added to existing API version is against Azure API guidelines. https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#123-definition-of-a-breaking-change

@hvermis Could you please help clarify ADF definitions for breaking changes to the whole team, otherwise we can't do anything to add new ADF connectors, thanks!

**Pull request closed without merging**